### PR TITLE
Improved .NET Framework version consistency and added test of Framework.Web

### PR DESCRIPTION
**Pull request by jwrosewell**
TEST:
Added a framework project test to verify that the framework dependencies work as expected when used with FiftyOne.Pipeline.Web.Framework. Required a change to the Framework web project to enable the function that resolves the base directory of the application to be replaceable.
Added a test helper to compare json strings by removing carriage returns and trailing and leading white space. Fixes some tests not always passing when indentation is considered.

UPDATE:
Target all .NET Framework projects to 4.6.2 as this the oldest version in support until January 2027. See: https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-framework
Updated all versions of dependencies across all the four solutions included in the repository.
Consolidated the package Framework references into .csproj or .config files and not used packages.config.
Downgraded the version of web infrastructure used by framework with the web as the other packages from Microsoft depend on v1 rather than v2 and this avoids version warnings.
Reduced the dependent packages to the minimum needed for the example to work.

CLEANUP:
Removed unused references.

### CODE REVIEW

File: FiftyOne.Pipeline.Web.sln [hunk 1]
```diff
@@ -67,170 +67,180 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "AspNet Framework MVC Exampl
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AspNetCore Example", "Web Integration\Examples\NetCore\AspNetCore Example.csproj", "{AF8687D1-E12A-430F-A616-FBB0EB5D92D6}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "FiftyOne.Pipeline.Web.Framework.Tests", "Web Integration\Tests\FiftyOne.Pipeline.Web.Framework.Tests\FiftyOne.Pipeline.Web.Framework.Tests.csproj", "{4E0D24CA-D682-4BFC-B506-31B9BE0DEA31}"
```
Comment (ben51degrees) [inline FiftyOne.Pipeline.Web.sln:70]: I suspect this will cause a problem with `dotnet build` as there's a Framework project in there.

File: Web Integration/Tests/FiftyOne.Pipeline.Web.Shared.Tests/FiftyOne.Pipeline.Web.Shared.Tests.csproj [hunk 2]
```diff
@@ -4,14 +4,18 @@
     <TargetFramework>net6.0</TargetFramework>
 
     <IsPackable>false</IsPackable>
+
+    <StartupObject></StartupObject>
+
+    <IsPublishable>True</IsPublishable>
```
Comment (ben51degrees) [inline Web Integration/Tests/FiftyOne.Pipeline.Web.Shared.Tests/FiftyOne.Pipeline.Web.Shared.Tests.csproj:10]: Is this addition necessary? I haven't looked into how this is used by dotnet, but it could result in the test package being released to NuGet.

File: FiftyOne.Pipeline.Web.sln (hunk 1 above)
Comment (jwrosewell) [reply FiftyOne.Pipeline.Web.sln:70]: The solution contains both Core and Framework so this new test should be included.

File: Web Integration/Tests/FiftyOne.Pipeline.Web.Shared.Tests/FiftyOne.Pipeline.Web.Shared.Tests.csproj (hunk 2 above)
Comment (jwrosewell) [reply Web Integration/Tests/FiftyOne.Pipeline.Web.Shared.Tests/FiftyOne.Pipeline.Web.Shared.Tests.csproj]: Removed these lines in commit just now.

**Review by ben51degrees — APPROVED** [review]
